homeassistant/components/gree/climate.py

```diff
@@ -0,0 +1,405 @@
+        return MAX_TEMP
+
+    @property
+    def target_temperature_step(self) -> float:
+        """Return the target temperature step support by the device."""
+        return TARGET_TEMPERATURE_STEP
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return the current HVAC mode for the device."""
+        if not self._device.power:
+            return HVAC_MODE_OFF
+
+        return HVAC_MODES.get(self._device.mode)
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set new target hvac mode."""
+        _LOGGER.debug(
+            "Setting HVAC mode to %s for device %s", hvac_mode, self._name,
+        )
+
+        if hvac_mode == HVAC_MODE_OFF:
+            self._device.power = False
+            return await self._push_state_update()
+
+        if not self._device.power:
+            self._device.power = True
+
+        self._device.mode = HVAC_MODES_REVERSE.get(hvac_mode)
+        return await self._push_state_update()
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the HVAC modes support by the device."""
+        modes = list(HVAC_MODES.values())
+        modes.append(HVAC_MODE_OFF)
+        return modes
+
+    @property
+    def preset_mode(self) -> str:
+        """Return the current preset mode for the device."""
+        if self._device.steady_heat:
+            return PRESET_AWAY
+        if self._device.power_save:
+            return PRESET_ECO
+        if self._device.sleep:
+            return PRESET_SLEEP
+        if self._device.turbo:
+            return PRESET_BOOST
+        return PRESET_NONE
+
+    async def async_set_preset_mode(self, preset_mode):
+        """Set new preset mode."""
+        _LOGGER.debug(
+            "Setting preset mode to %s for device %s", preset_mode, self._name,
+        )
+
+        if preset_mode not in PRESET_MODES:
+            _LOGGER.warning("Received and invalid preset mode: %s", preset_mode)
+            return
+
+        self._device.steady_heat = False
+        self._device.power_save = False
+        self._device.turbo = False
+        self._device.sleep = False
+
+        if preset_mode == PRESET_AWAY:
+            self._device.steady_heat = True
+        elif preset_mode == PRESET_ECO:
+            self._device.power_save = True
+        elif preset_mode == PRESET_BOOST:
+            self._device.turbo = True
+        elif preset_mode == PRESET_SLEEP:
+            self._device.sleep = True
+
+        return await self._push_state_update()
+
+    @property
+    def preset_modes(self) -> List[str]:
+        """Return the preset modes support by the device."""
+        return PRESET_MODES
+
+    @property
+    def fan_mode(self) -> str:
+        """Return the current fan mode for the device."""
+        speed = self._device.fan_speed
+        return FAN_MODES.get(speed)
+
+    async def async_set_fan_mode(self, fan_mode):
+        """Set new target fan mode."""
+        if fan_mode not in FAN_MODES_REVERSE.keys():
```

Comment:
The default iterator of a dict is the keys.
```suggestion
        if fan_mode not in FAN_MODES_REVERSE:
```